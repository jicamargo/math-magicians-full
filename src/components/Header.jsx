import styles from "@/styles/Header.module.css";
import PropTypes from "prop-types";

const Header = (props) => {
  return (
    <header className={styles.header}>
       {props.children}
    </header>
  );
};

Header.defaultProps = {
  children: null,
};

Header.propTypes = {
  children: PropTypes.node,
};

export default Header;
