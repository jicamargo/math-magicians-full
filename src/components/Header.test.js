// eslint-disable-next-line no-unused-vars
import React from 'react';
import { render, screen } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import Header from './Header';

describe('Header', () => {
  it('renders the Header component', () => {
    render(<Header>Test Header</Header>);
    expect(screen.getByRole('banner')).toMatchSnapshot();
  });

  it('renders the header class correctly', () => {
    render(<Header>Test Header</Header>);
    const headerElement = screen.getByRole('banner');
    expect(headerElement).toHaveClass('header');
  });

  it('renders the children correctly', () => {
    render(<Header>Test Header</Header>);
    expect(screen.getByText('Test Header')).toBeInTheDocument();
  });
});
