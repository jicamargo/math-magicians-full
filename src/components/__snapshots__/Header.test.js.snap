// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header renders the Header component 1`] = `
<header
  class="header"
>
  Test Header
</header>
`;
